portfolio/src/main/java/com/google/sps/servlets/UserServlet.java

```diff
@@ -0,0 +1,114 @@
+import com.google.sps.data.Log;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+
+
+@WebServlet("/user")
+public class UserServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("Log");
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    UserService userService = UserServiceFactory.getUserService();
+
+    
+    
+    List<Log> Logs = new ArrayList<>();
+    String status;
+    String userEmail;
+    String urlToRedirectToAfterUserLogsIn;
+    String urlToRedirectToAfterUserLogsOut;
+    String loginUrl;
+    String logoutUrl;
+
+
+    for (Entity entity : results.asIterable()) {//iterate through the logs
+        //get all the properties
+        if (userService.isUserLoggedIn()) {
+          status = "In";
+          userEmail = userService.getCurrentUser().getEmail();
+          urlToRedirectToAfterUserLogsIn = "/";
+          urlToRedirectToAfterUserLogsOut = "/";
+          loginUrl = "";
+          logoutUrl = userService.createLogoutURL(urlToRedirectToAfterUserLogsOut);
+          
+        } 
+        else {
+          status = "Out";
+          urlToRedirectToAfterUserLogsIn = "/";
+          urlToRedirectToAfterUserLogsOut = "/";
+          userEmail = "";
+          loginUrl = userService.createLoginURL(urlToRedirectToAfterUserLogsIn);
+          logoutUrl = "";
+       
+        }
+
+        //create new log object
+        Log log = new Log(status, loginUrl, logoutUrl, userEmail);
+        //add to Logs
+         Logs.add(log);
+         
+         
+    }
+
+
+    Gson gson = new Gson();
+    response.setContentType("application/json;");
+    response.getWriter().println(gson.toJson(Logs));
+  }
+
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String status = request.getParameter("status");
+    String loginUrl = request.getParameter("loginUrl");
+    String logoutUrl = request.getParameter("logoutUrl");
+    String userEmail = request.getParameter("userEmail");
+
+    Entity logEntity = new Entity("Log");
+   
+    logEntity.setProperty("status", status);
+    logEntity.setProperty("loginUrl", loginUrl);
+    logEntity.setProperty("logoutUrl", logoutUrl);
+    logEntity.setProperty("userEmail", userEmail);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(logEntity);
+
+    //response.sendRedirect("/index.html");
+  }
```

Comment:
Also maybe don't need a POST action since there's nothing that you need the server to save/consume for you.